File: src/jsonrpc/eth/accounts/eth_accounts.js
/**
 * @fileoverview eth_accounts JSON-RPC method
 */

/**
 * @typedef {import('../../types/index.js').AddressType} Address
 * @typedef {import('../../types/index.js').Hash} Hash
 * @typedef {import('../../types/index.js').Quantity} Quantity
 * @typedef {import('../../types/index.js').BlockTag} BlockTag
 * @typedef {import('../../types/index.js').BlockSpec} BlockSpec
 * @typedef {import('../../../provider/types.js').RequestArguments} RequestArguments
 */

/**
 * Returns a list of addresses owned by client.
 *
 * @example
 * Result: ...
 *
 * Implements the `eth_accounts` JSON-RPC method.
 */

/** The JSON-RPC method name */
export const method = "eth_accounts";
/**
 * Result for `eth_accounts`
 *
 * Accounts
 *
 * @typedef {Quantity} Result
 */

/**
 * Creates a eth_accounts JSON-RPC request
 *
 * @returns {RequestArguments}
 */
export function AccountsRequest() {
	return { method };
}

File: packages/voltaire-ts/src/jsonrpc/eth/accounts/eth_accounts.go
package eth

import (
	"encoding/json"
	"fmt"

	"github.com/ethereum/execution-apis/types"
)

// Returns a list of addresses owned by client.
//
// Example:
// Result: ...
//
// Implements the eth_accounts JSON-RPC method.

// Method is the JSON-RPC method name
const Method = "eth_accounts"

// Params represents the parameters for eth_accounts
type Params struct {
}

// MarshalJSON implements json.Marshaler for Params.
// JSON-RPC 2.0 uses positional array parameters.
func (p Params) MarshalJSON() ([]byte, error) {
	return json.Marshal([]interface{}{
	})
}

// UnmarshalJSON implements json.Unmarshaler for Params.
func (p *Params) UnmarshalJSON(data []byte) error {
	var arr []json.RawMessage
	if err := json.Unmarshal(data, &arr); err != nil {
		return err
	}

	if len(arr) != 0 {
		return fmt.Errorf("expected 0 parameters, got %d", len(arr))
	}

	return nil
}

// Result represents the result for eth_accounts
//
// Accounts
type Result struct {
	Value types.Quantity `json:"-"`
}

// MarshalJSON implements json.Marshaler for Result.
func (r Result) MarshalJSON() ([]byte, error) {
	return json.Marshal(r.Value)
}

// UnmarshalJSON implements json.Unmarshaler for Result.
func (r *Result) UnmarshalJSON(data []byte) error {
	return json.Unmarshal(data, &r.Value)
}
